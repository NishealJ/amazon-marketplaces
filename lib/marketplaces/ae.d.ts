import { AmazonMarketplace } from '../amazon-marketplace';
export declare const AE: AmazonMarketplace;
