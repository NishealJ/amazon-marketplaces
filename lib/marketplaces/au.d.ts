import { AmazonMarketplace } from '../amazon-marketplace';
export declare const AU: AmazonMarketplace;
