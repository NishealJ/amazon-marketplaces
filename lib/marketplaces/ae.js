"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AE = void 0;
const amazon_marketplace_1 = require("../amazon-marketplace");
const marketplace_advertising_regions_1 = require("../marketplace-advertising-regions");
const selling_partner_api_regions_1 = require("../selling-partner-api-regions");
exports.AE = new amazon_marketplace_1.AmazonMarketplace({
    countryCode: amazon_marketplace_1.AmazonMarketplaceCountryCode.AE,
    timeZone: amazon_marketplace_1.AmazonMarketplaceTimeZone.ASIA_DUBAI,
    currency: amazon_marketplace_1.AmazonMarketplaceAdvertisingCurrency.AED,
    id: 'A2VIGQ35RCS4UG',
    name: 'United Arab Emirates',
    uri: 'https://www.amazon.ae',
    sellerCentralUri: 'https://sellercentral.amazon.ae',
    vendorCentralUri: 'https://vendorcentral.amazon.me',
    webServiceUri: 'https://mws.amazonservices.ae',
    launchDate: '2019-05-01',
    advertising: {
        countryCode: amazon_marketplace_1.AmazonMarketplaceAdvertisingCountryCode.AE,
        region: marketplace_advertising_regions_1.marketplaceAdvertisingRegions.EU,
        bids: {
            sponsoredBrands: {
                min: 40,
                max: 18400,
            },
            sponsoredProducts: {
                min: 24,
                max: 18400,
            },
        },
        timeZone: amazon_marketplace_1.AmazonMarketplaceAdvertisingTimeZone.ASIA_DUBAI,
    },
    sellingPartner: {
        region: selling_partner_api_regions_1.sellingPartnerRegions.EU,
    },
});
//# sourceMappingURL=ae.js.map