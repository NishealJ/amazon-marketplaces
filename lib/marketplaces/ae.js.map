{"version":3,"file":"ae.js","sourceRoot":"","sources":["../../src/marketplaces/ae.ts"],"names":[],"mappings":";;;AAAA,8DAO8B;AAC9B,wFAAkF;AAClF,gFAAsE;AAEzD,QAAA,EAAE,GAAG,IAAI,sCAAiB,CAAC;IACtC,WAAW,EAAE,iDAA4B,CAAC,EAAE;IAC5C,QAAQ,EAAE,8CAAyB,CAAC,UAAU;IAC9C,QAAQ,EAAE,yDAAoC,CAAC,GAAG;IAClD,EAAE,EAAE,gBAAgB;IACpB,IAAI,EAAE,sBAAsB;IAC5B,GAAG,EAAE,uBAAuB;IAC5B,gBAAgB,EAAE,iCAAiC;IACnD,gBAAgB,EAAE,iCAAiC;IACnD,aAAa,EAAE,+BAA+B;IAC9C,UAAU,EAAE,YAAY;IACxB,WAAW,EAAE;QACX,WAAW,EAAE,4DAAuC,CAAC,EAAE;QACvD,MAAM,EAAE,+DAA6B,CAAC,EAAE;QACxC,IAAI,EAAE;YACJ,eAAe,EAAE;gBACf,GAAG,EAAE,EAAE;gBACP,GAAG,EAAE,KAAM;aACZ;YACD,iBAAiB,EAAE;gBACjB,GAAG,EAAE,EAAE;gBACP,GAAG,EAAE,KAAM;aACZ;SACF;QACD,QAAQ,EAAE,yDAAoC,CAAC,UAAU;KAC1D;IACD,cAAc,EAAE;QACd,MAAM,EAAE,mDAAqB,CAAC,EAAE;KACjC;CACF,CAAC,CAAA"}