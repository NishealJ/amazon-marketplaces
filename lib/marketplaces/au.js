"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AU = void 0;
const amazon_marketplace_1 = require("../amazon-marketplace");
const marketplace_advertising_regions_1 = require("../marketplace-advertising-regions");
const selling_partner_api_regions_1 = require("../selling-partner-api-regions");
exports.AU = new amazon_marketplace_1.AmazonMarketplace({
    countryCode: amazon_marketplace_1.AmazonMarketplaceCountryCode.AU,
    timeZone: amazon_marketplace_1.AmazonMarketplaceTimeZone.AUSTRALIA_SYDNEY,
    currency: amazon_marketplace_1.AmazonMarketplaceAdvertisingCurrency.AUD,
    id: 'A39IBJ37TRP1C6',
    name: 'Australia',
    uri: 'https://www.amazon.com.au',
    sellerCentralUri: 'https://sellercentral.amazon.com.au',
    vendorCentralUri: 'https://vendorcentral.amazon.com.au',
    webServiceUri: 'https://mws.amazonservices.com.au',
    launchDate: '2017-11-01',
    advertising: {
        countryCode: amazon_marketplace_1.AmazonMarketplaceAdvertisingCountryCode.AU,
        region: marketplace_advertising_regions_1.marketplaceAdvertisingRegions.FE,
        bids: {
            sponsoredBrands: {
                min: 10,
                max: 7000,
            },
            sponsoredProducts: {
                min: 10,
                max: 141000,
            },
        },
        timeZone: amazon_marketplace_1.AmazonMarketplaceAdvertisingTimeZone.AUSTRALIA_SYDNEY,
    },
    sellingPartner: {
        region: selling_partner_api_regions_1.sellingPartnerRegions.FE,
    },
});
//# sourceMappingURL=au.js.map