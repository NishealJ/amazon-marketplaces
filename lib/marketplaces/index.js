"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.amazonMarketplacesList = exports.amazonMarketplaces = void 0;
const ae_1 = require("./ae");
const au_1 = require("./au");
const be_1 = require("./be");
const br_1 = require("./br");
const ca_1 = require("./ca");
const cn_1 = require("./cn");
const de_1 = require("./de");
const eg_1 = require("./eg");
const es_1 = require("./es");
const fr_1 = require("./fr");
const gb_1 = require("./gb");
const in_1 = require("./in");
const it_1 = require("./it");
const jp_1 = require("./jp");
const mx_1 = require("./mx");
const nl_1 = require("./nl");
const pl_1 = require("./pl");
const sa_1 = require("./sa");
const se_1 = require("./se");
const sg_1 = require("./sg");
const tr_1 = require("./tr");
const us_1 = require("./us");
exports.amazonMarketplaces = {
    /**
     * United Arab Emirates
     */
    AE: ae_1.AE,
    /**
     * Australia
     */
    AU: au_1.AU,
    /**
     * Belgium
     */
    BE: be_1.BE,
    /**
     * Brazil
     */
    BR: br_1.BR,
    /**
     * Canada
     */
    CA: ca_1.CA,
    /**
     * China
     */
    CN: cn_1.CN,
    /**
     * Germany
     */
    DE: de_1.DE,
    /**
     * Egypt
     */
    EG: eg_1.EG,
    /**
     * Spain
     */
    ES: es_1.ES,
    /**
     * France
     */
    FR: fr_1.FR,
    /**
     * United Kingdom
     */
    GB: gb_1.GB,
    /**
     * India
     */
    IN: in_1.IN,
    /**
     * Italy
     */
    IT: it_1.IT,
    /**
     * Japan
     */
    JP: jp_1.JP,
    /**
     * Mexico
     */
    MX: mx_1.MX,
    /**
     * Netherlands
     */
    NL: nl_1.NL,
    /**
     * Poland
     */
    PL: pl_1.PL,
    /**
     * Saudi Arabia
     */
    SA: sa_1.SA,
    /**
     * Sweden
     */
    SE: se_1.SE,
    /**
     * Singapore
     */
    SG: sg_1.SG,
    /**
     * Turkey
     */
    TR: tr_1.TR,
    /**
     * United States
     */
    US: us_1.US,
};
/**
 * An array of all of the `AmazonMarketplace` objects.
 */
exports.amazonMarketplacesList = Object.values(exports.amazonMarketplaces);
//# sourceMappingURL=index.js.map