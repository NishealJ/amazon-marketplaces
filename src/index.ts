export * from './marketplaces'
export * from './amazon-marketplace'
