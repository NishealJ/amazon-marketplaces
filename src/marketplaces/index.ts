import { AE } from './ae'
import { AU } from './au'
import { BR } from './br'
import { CA } from './ca'
import { CN } from './cn'
import { DE } from './de'
import { EG } from './eg'
import { ES } from './es'
import { FR } from './fr'
import { GB } from './gb'
import { IN } from './in'
import { IT } from './it'
import { JP } from './jp'
import { MX } from './mx'
import { NL } from './nl'
import { SA } from './sa'
import { SG } from './sg'
import { TR } from './tr'
import { US } from './us'

export const amazonMarketplaces = {
  /**
   * United Arab Emirates
   */
  AE,

  /**
   * Australia
   */
  AU,

  /**
   * Brazil
   */
  BR,

  /**
   * Canada
   */
  CA,

  /**
   * China
   */
  CN,

  /**
   * Germany
   */
  DE,

  /**
   * Egypt
   */
  EG,

  /**
   * Spain
   */
  ES,

  /**
   * France
   */
  FR,

  /**
   * United Kingdom
   */
  GB,

  /**
   * India
   */
  IN,

  /**
   * Italy
   */
  IT,

  /**
   * Japan
   */
  JP,

  /**
   * Mexico
   */
  MX,

  /**
   * Netherlands
   */
  NL,

  /**
   * Singapore
   */
  SA,

  /**
   * Singapore
   */
  SG,

  /**
   * Turkey
   */
  TR,

  /**
   * United States
   */
  US,
} as const

/**
 * An array of all of the `AmazonMarketplace` objects.
 */
export const amazonMarketplacesList = Object.values(amazonMarketplaces)
