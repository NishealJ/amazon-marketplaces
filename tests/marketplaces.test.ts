import { amazonMarketplaces } from '../src/marketplaces'
import { readdirSync } from 'fs'
import { join } from 'path'

const countryCodes = readdirSync(join(__dirname, '../src/marketplaces'))
  .map(f => f.split('.')[0])
  .filter(f => f.match(/^[A-Z]{2}$/))

describe('marketplace', () => {
  it('has a known number of marketplaces', () => {
    expect.assertions(1)

    expect(countryCodes.length).toBe(17)
  })

  it('files should match the exports', () => {
    expect.assertions(countryCodes.length)

    countryCodes.forEach(countryCode => {
      const code = (countryCode as unknown) as keyof typeof amazonMarketplaces
      expect(amazonMarketplaces[code].countryCode).toBe(code)
    })
  })

  it('object is read only', () => {
    expect(() => Object.assign(amazonMarketplaces.CA, { id: 'a' })).toThrowError(/Cannot assign to read only property/)
  })
})
