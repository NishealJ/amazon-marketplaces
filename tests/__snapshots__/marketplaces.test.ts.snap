// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marketplace AE should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 18400,
        "min": 24,
      },
      "sponsoredProducts": Object {
        "max": 18400,
        "min": 24,
      },
    },
    "countryCode": "AE",
    "region": "EU",
    "timeZone": "Asia/Dubai",
    "uri": "https://advertising-api-eu.amazon.com",
  },
  "countryCode": "AE",
  "currency": "AED",
  "id": "A2VIGQ35RCS4UG",
  "name": "United Arab Emirates",
  "uri": "https://www.amazon.ae",
  "webServiceUri": "https://mws.amazonservices.ae",
}
`;

exports[`marketplace AU should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 7000,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 141000,
        "min": 10,
      },
    },
    "countryCode": "AU",
    "region": "FE",
    "timeZone": "Australia/Sydney",
    "uri": "https://advertising-api-fe.amazon.com",
  },
  "countryCode": "AU",
  "currency": "AUD",
  "id": "A39IBJ37TRP1C6",
  "name": "Australia",
  "uri": "https://www.amazon.com.au",
  "webServiceUri": "https://mws.amazonservices.com.au",
}
`;

exports[`marketplace BR should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "BR",
  "currency": "BRL",
  "id": "A2Q3Y263D00KWC",
  "name": "Brazil",
  "uri": "https://www.amazon.com.br",
  "webServiceUri": "https://mws.amazonservices.com",
}
`;

exports[`marketplace CA should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 4900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "CA",
    "region": "NA",
    "timeZone": "America/Los_Angeles",
    "uri": "https://advertising-api.amazon.com",
  },
  "countryCode": "CA",
  "currency": "CAD",
  "id": "A2EUQ1WTGCTBG2",
  "name": "Canada",
  "uri": "https://www.amazon.ca",
  "webServiceUri": "https://mws.amazonservices.ca",
}
`;

exports[`marketplace CN should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "CN",
  "currency": "CNY",
  "id": "AAHKV2X7AFYLW",
  "name": "China",
  "uri": "https://www.amazon.cn",
  "webServiceUri": "https://mws.amazonservices.com.cn",
}
`;

exports[`marketplace DE should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "DE",
    "region": "EU",
    "timeZone": "Europe/Paris",
    "uri": "https://advertising-api-eu.amazon.com",
  },
  "countryCode": "DE",
  "currency": "EUR",
  "id": "A1PA6795UKMFR9",
  "name": "Germany",
  "uri": "https://www.amazon.de",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace ES should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "ES",
    "region": "EU",
    "timeZone": "Europe/Paris",
    "uri": "https://advertising-api-eu.amazon.com",
  },
  "countryCode": "ES",
  "currency": "EUR",
  "id": "A1RKKUPIHCS9HS",
  "name": "Spain",
  "uri": "https://www.amazon.es",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace FR should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "FR",
    "region": "EU",
    "timeZone": "Europe/Paris",
    "uri": "https://advertising-api-eu.amazon.com",
  },
  "countryCode": "FR",
  "currency": "EUR",
  "id": "A13V1IB3VIYZZH",
  "name": "France",
  "uri": "https://www.amazon.fr",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace GB should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3100,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "UK",
    "region": "EU",
    "timeZone": "Europe/London",
    "uri": "https://advertising-api-eu.amazon.com",
  },
  "countryCode": "GB",
  "currency": "GBP",
  "id": "A1F83G8C2ARO7P",
  "name": "United Kingdom",
  "uri": "https://www.amazon.co.uk",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace IN should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "IN",
  "currency": "INR",
  "id": "A21TJRUUN4KGV",
  "name": "India",
  "uri": "https://www.amazon.in",
  "webServiceUri": "https://mws.amazonservices.in",
}
`;

exports[`marketplace IT should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 3900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "DE",
    "region": "EU",
    "timeZone": "Europe/Paris",
    "uri": "https://advertising-api-eu.amazon.com",
  },
  "countryCode": "IT",
  "currency": "EUR",
  "id": "APJ6JRA9NG5V4",
  "name": "Italy",
  "uri": "https://www.amazon.it",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace JP should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 776000,
        "min": 1000,
      },
      "sponsoredProducts": Object {
        "max": 10000000,
        "min": 200,
      },
    },
    "countryCode": "JP",
    "region": "FE",
    "timeZone": "Asia/Tokyo",
    "uri": "https://advertising-api-fe.amazon.com",
  },
  "countryCode": "JP",
  "currency": "JPY",
  "id": "A1VC38T7YXB528",
  "name": "Japan",
  "uri": "https://www.amazon.co.jp",
  "webServiceUri": "https://mws.amazonservices.jp",
}
`;

exports[`marketplace MX should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "MX",
  "currency": "MXN",
  "id": "A1AM78C64UM0Y8",
  "name": "Mexico",
  "uri": "https://www.amazon.com.mx",
  "webServiceUri": "https://mws.amazonservices.com.mx",
}
`;

exports[`marketplace NL should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "NL",
  "currency": "EUR",
  "id": "A1805IZSGTT6HS",
  "name": "Netherlands",
  "uri": "https://www.amazon.nl",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace SG should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "SG",
  "currency": "SGD",
  "id": "A19VAU5U5O7RUS",
  "name": "Singapore",
  "uri": "https://www.amazon.sg",
  "webServiceUri": "https://mws-fe.amazonservices.com",
}
`;

exports[`marketplace TR should match snapshot 1`] = `
AmazonMarketplace {
  "countryCode": "TR",
  "currency": "TRY",
  "id": "A33AVAJ2PDY3EV",
  "name": "Turkey",
  "uri": "https://www.amazon.com.tr",
  "webServiceUri": "https://mws-eu.amazonservices.com",
}
`;

exports[`marketplace US should match snapshot 1`] = `
AmazonMarketplace {
  "advertising": Object {
    "bids": Object {
      "sponsoredBrands": Object {
        "max": 4900,
        "min": 10,
      },
      "sponsoredProducts": Object {
        "max": 100000,
        "min": 2,
      },
    },
    "countryCode": "US",
    "region": "NA",
    "timeZone": "America/Los_Angeles",
    "uri": "https://advertising-api.amazon.com",
  },
  "countryCode": "US",
  "currency": "USD",
  "id": "ATVPDKIKX0DER",
  "name": "United States",
  "uri": "https://www.amazon.com",
  "webServiceUri": "https://mws.amazonservices.com",
}
`;
